### ISSUE TITLE
Spiky mesh for gltf models with skeleton

### ISSUE THREAD

**Issue by skywalkershen** [iTowns/itowns#1866]
## Your Environment

* Version used: 2.83.2
* Browser Name and version:Chrome/102.0.0.0
* Operating System and version (desktop or mobile):Intel Mac OS X 10_15_7
* Link to your project: Currently not available

## Context

I'm migrating from pure threejs scene to itowns, aiming to load gltf models with bones, then adjust bone rotations to visualize result of motion detection.
Yet, the working code in pure threejs scene is not working in itowns: when loading gltf, it renders boned models with spiky mesh, and my vscode/chrome not responding on model loaded for several seconds, sometimes crash.

## Steps to Reproduce (for bugs)

1. Download model from https://www.mixamo.com/#/?page=1&type=Character, I tested with "The Boss" and "CLARIE".
2. Export a boned gltf with blender.
3. Remove bone, export a gltf without bone using blender(delete "Armature")
4. Code for loading gltf:
```
// import { Coordinates, legacyGLTFLoader as GLTFLoader} from 'itowns'
import { Coordinates, glTFLoader as GLTFLoader} from 'itowns'
const models = new Map(),
    gltfLoader = GLTFLoader.constructor();

loadGltfModel(testModelName, testModelUrl)
async function loadGltfModel (name, url, rotateOnAxisParam, scale=1) {
    if (!name) return console.error('Must provide name for the model');
    if (!url) return console.error('Must provide url for the model')
    return new Promise((resolve, reject) => {
        gltfLoader.load(
            url,
            gltf => {
                let model = gltf.scene,
                    coord = new Coordinates('EPSG:4326', 116, 39, 0),
                    gltfID = view.mainLoop.gfxEngine.getUniqueThreejsLayer();

                model.position.copy(coord.as(view.referenceCrs));;
                // align up vector with geodesic normal
                // model.lookAt(model.position.clone().add(coord.geodesicNormal));
                // // user rotate building to align with ortho image
                // model.rotateX(Math.PI * 0.5);
                // set camera's layer to do not disturb the picking
                // model.traverse(function _(obj) { obj.layers.set(gltfID); });
                view.camera.camera3D.layers.enable(gltfID);
                
                model.scale.set(scale, scale, scale);
                model.updateMatrixWorld();
                model.castShadow = true;
                model.receiveShadow = true;
                // model.traverse( function ( child ) {
                //     if ( child.isMesh ) {
                //             child.castShadow = true;
                //             child.receiveShadow = true;
                //     }
                // } );

                // if (model.children[0]) {
                //     const helper = new SkeletonHelper( model.children[0]);
                //     model.skeletonHelper = helper;
                //     view.scene.add( helper );
                // }
                if (Array.isArray(rotateOnAxisParam)) {
                    rotateOnAxisParam.forEach(param => {
                        rotateOnAxis(model, param);
                    })
                }
                view.scene.add(model);
                view.notifyChange();
                models.set(name, model);
                
                resolve(model);
            },
            (xhr) => {
                console.debug(`Model ${name} loaded: ${(xhr.loaded / xhr.total) * 100}%`)
            },
            (error) => {
                console.error(error)
                reject(error);
            }
        )
    })
```

## Expected Behavior

models with or without bones should look the same.

## Actual Behavior

![image](https://user-images.githubusercontent.com/32822916/174757243-692c95dd-5e52-4d1a-8af7-12dbae3f3d29.png)
test model "The Boss" without bones
![image](https://user-images.githubusercontent.com/32822916/174757456-987c29db-7a74-46f7-be38-976d970af398.png)
test model "The Boss" with bones

## Possible Cause/Fix/Solution

I noticed the skinnedMesh is removed, is it the cause of the issue? https://github.com/iTowns/itowns/pull/956
I tried to use the " legacyGLTFLoader" exposed by itowns but it returned an error
![image](https://user-images.githubusercontent.com/32822916/174758323-30716610-3e71-46db-b647-b91115a1a270.png)

**Comment by jailln** [comment]
Hi @skywalkershen, sorry for the late reply.
Which version of threejs are you using in your pure threejs scene?

**Comment by skywalkershen** [comment]
> Hi @skywalkershen, sorry for the late reply. Which version of threejs are you using in your pure threejs scene?

Hi, I'm still troubled with this issue and there are some updates. What I'm trying to do is to control the body move and face expression of my model, I managed to do that with pure threejs, yet things get weird with itowns:

The threejs version I'm using is 0.140.2", I also tried using this version for itowns for gltf loaders.

1. I found out the distorted skinnedMesh issue might have something to do with scale.
Taking model from this threejs example as an example: [threejs example](https://github.com/mrdoob/three.js/blob/master/examples/webgl_animation_skinning_morph.html)
when scaling to 1, it gets distorted, but not that much
<img width="347" alt="image" src="https://user-images.githubusercontent.com/32822916/197559370-956ee67a-fa56-4b0d-97bf-9269ed653a91.png">
when scaling to 100, it looks normal
<img width="376" alt="image" src="https://user-images.githubusercontent.com/32822916/197559476-9c940cd6-b860-4a63-b5a1-7790eb0335dd.png">

A more obvious example using [model from this threejs example](url)
distorted when scaling to 1: 
<img width="505" alt="image" src="https://user-images.githubusercontent.com/32822916/197560012-e5bf0a5c-7404-40d9-950f-47518cce2e1d.png">
better when scaling to 100:
<img width="591" alt="image" src="https://user-images.githubusercontent.com/32822916/197560247-a3cc6279-a061-4041-97d0-4e3fa02f6363.png">
And for all other models either from threejs examples or other sources, the boneless mesh has no problem at any scale level, but the skinnedMesh always gets distorted at small scale, yet it is the scale I need to work with.

2. I' also trying to use morph targets for , but after importing my gltf models, the structure seems to be changed and the node with morph targets got lost.
still using [model from threejs example](https://threejs.org/examples/?q=morph#webgl_animation_skinning_morph):
The structure of the model supposed to be:
The head_4 is the node with morph targets
<img width="399" alt="image" src="https://user-images.githubusercontent.com/32822916/197561583-ea99014c-953f-4303-8f9a-0f9c42f1b895.png">

Yet when loaded in itowns, the object structure goes totally different: 
<img width="648" alt="image" src="https://user-images.githubusercontent.com/32822916/197562061-91dfb2b8-9506-4c2e-93c7-94bc4cb3473c.png">
The head_4 is no longer there, there is no way I can get access to the morph targets.

I didn't really look into source code of itowns, but I noticed that important components like renderer, scene, skinnedMesh, Mesh are directly taken from threejs, so I wonder what's the cause of this issue, and whether there is a workaround.

Thank you

**Comment by jailln** [comment]
I'm not used to animated models but isn't there anything else to do when the gltf is loaded? (e.g. adding animations to scene.animations ?) I suppose you use the same code in itowns and threejs to add your gltf model to the scene?

Did you try with threejs GLTFLoader instead of itowns one ? We patch the GLTFLoader in itowns to support a specific property of 3DTiles that are not in the gltf standard. The patch is [here](https://github.com/iTowns/itowns/blob/master/config/replace.config.mjs), maybe that's the cause of the issue?

Let me know how that works out :)

**Comment by skywalkershen** [comment]
> I'm not used to animated models but isn't there anything else to do when the gltf is loaded? (e.g. adding animations to scene.animations ?) I suppose you use the same code in itowns and threejs to add your gltf model to the scene?
> 
> Did you try with threejs GLTFLoader instead of itowns one ? We patch the GLTFLoader in itowns to support a specific property of 3DTiles that are not in the gltf standard. The patch is [here](https://github.com/iTowns/itowns/blob/master/config/replace.config.mjs), maybe that's the cause of the issue?
> 
> Let me know how that works out :)

Yes, I use same code for itowns and pure threejs, and clearly it does not have anything to do with my logic, simply loading the model will cause the issue. And I don't think it has anything to do with animations either. I simply need skeleton and bone nodes, and none of the models I load comes with animation(my coworker made those boned-models). For pure threejs version, I just load the model with gltf loader, then getNode by their name and rotate it, that works fine.

And I did try importing gltf loader directly from three, same issue occurs. Please see jsfiddles for issue reproduce, both using threejs v146, with same cdn and model.
[Minimum example to load boned gltf model in itowns](http://jsfiddle.net/ue90sk56/3/), by changing model scale, you can see the model gets more and more spiky as it scales to 1.
screenshot for scaling to 10
![image](https://user-images.githubusercontent.com/32822916/200234837-0ab297d0-d59c-4e90-9e52-44b18cd4017d.png)

[Loading same model with pure threejs](https://jsfiddle.net/gakd93vh/1/), it just renders expected at any scale level.
![image](https://user-images.githubusercontent.com/32822916/200235058-316aad23-4ca2-47fc-9d7d-733d3586d253.png)
